src/librustc_ast/ast.rs

```diff
@@ -899,6 +899,33 @@ impl Stmt {
             _ => false,
         }
     }
+
+    pub fn is_expr_with<F>(&self, f: F) -> bool
+    where
+        F: FnOnce(&Expr) -> bool,
+    {
+        match self.kind {
+            StmtKind::Expr(ref e) => f(e),
+            _ => false,
+        }
+    }
+
+    pub fn is_semi_expr(&self) -> bool {
+        match self.kind {
+            StmtKind::Semi(_) => true,
+            _ => false,
+        }
+    }
+
```

Comment:
```suggestion
```
Seems like you ended up not needing this method afaict

Reply:
Yeah i just added it for parity with `StmtKind::Expr`